### PR TITLE
Issue #1600: Metric for total number of invoked calls from retry

### PR DESCRIPTION
**Pull request by laviua**
https://github.com/resilience4j/resilience4j/issues/1600

PR adds a totalAttemptCounter for success and retry attempts

### CODE REVIEW

**Review by flipch — APPROVED** [review]

**Comment by laviua** [comment]
@flipch could you please assist, what are the next steps?

**Comment by RobWin** [comment]
You also need to enhance `onComplete` for functions which don't return anything.

**Comment by laviua** [comment]
> You also need to enhance `onComplete` for functions which don't return anything.

@RobWin I've updated the implementation

File: resilience4j-metrics/src/test/java/io/github/resilience4j/metrics/RetryMetricsTest.java [hunk 1]
```diff
@@ -58,6 +58,23 @@ public void shouldReturnTotalNumberOfRequestsAs1ForSuccess() {
...
+
+        Retry retry = Retry.of("metrics", RetryConfig.<String>custom()
+                .retryOnResult(String::isEmpty)
+                .maxAttempts(5)
+                .build());
+
+
+        retry.executeRunnable(helloWorldService::sayHelloWorld);
+
+        //then(helloWorldService).should().sayHelloWorld();
```
Comment (andrzejsydor) [inline resilience4j-metrics/src/test/java/io/github/resilience4j/metrics/RetryMetricsTest.java:73]: why is it commented on, should be?

Comment (laviua) [reply resilience4j-metrics/src/test/java/io/github/resilience4j/metrics/RetryMetricsTest.java]: o, yes, pardon, my eyes got used to this line and I forgot to remove it

**Review by RobWin — APPROVED** [review]

**Pull request merged**